drivers/segger/console_rtt.c

```diff
@@ -0,0 +1,127 @@
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <unistd.h>
+#include <debug.h>
+#include <nuttx/fs/fs.h>
+
+#include <SEGGER_RTT.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+/* Character driver methods */
+
+static ssize_t rtt_read(FAR struct file *filep,
+                        FAR char *buffer,
+                        size_t buflen);
+static ssize_t rtt_write(FAR struct file *filep,
+                         FAR const char *buffer,
+                         size_t buflen);
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static const struct file_operations g_rttops =
+{
+  NULL,       /* open */
+  NULL,       /* close */
+  rtt_read,   /* read */
+  rtt_write,  /* write */
+  NULL,       /* seek */
+  NULL,       /* ioctl */
+  NULL        /* poll */
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+  , NULL      /* unlink */
+#endif
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: rtt_read
+ ****************************************************************************/
+
+static ssize_t rtt_read(FAR struct file *filep,
+                        FAR char *buffer, size_t buflen)
+{
+  ssize_t res = 0;
+
+  res += SEGGER_RTT_Read(0, buffer + res, buflen);
```

Comment:
you need follow the core logic of serial driver to compatible with most terminal. So please reference:
https://github.com/apache/incubator-nuttx/blob/master/drivers/serial/serial.c or
https://github.com/apache/incubator-nuttx/blob/master/drivers/serial/pty.c